src/Shared/PackageManagers/NuGetProjectManager.cs

```diff
@@ -337,267 +214,210 @@ public override Uri GetPackageAbsoluteUri(PackageURL purl)
-                        {
-                            Repository repository = new()
-                            {
-                                Type = "github"
-                            };
-                        
-                            await repository.ExtractRepositoryMetadata(githubPurl!);
+            // TODO: size and hash
 
-                            metadata.Repository ??= new List<Repository>();
-                            metadata.Repository.Add(repository);
-                        }
-                    }
-                    
-                    // dependencies
-                    IList<PackageDependencyGroup>? dependencyGroups = nuspecReader?.GetDependencyGroups().ToList();
-                    if (dependencyGroups is not null && dependencyGroups.Any())
-                    {
-                        metadata.Dependencies ??= new List<Dependency>();
+            // Homepage url
+            metadata.Homepage = packageVersion.ProjectUrl?.ToString();
 
-                        foreach (PackageDependencyGroup dependencyGroup in dependencyGroups)
-                        {
-                            dependencyGroup.Packages.ToList().ForEach((dependency) => metadata.Dependencies.Add(new Dependency() { Package = dependency.ToString(), Framework = dependencyGroup.TargetFramework.ToString()}));
-                        }
-                    }
+            // Authors and Maintainers
+            UpdateMetadataAuthorsAndMaintainers(metadata, packageVersion);
 
-                    // keywords
-                    metadata.Keywords = OssUtilities.ConvertJSONToList(OssUtilities.GetJSONPropertyIfExists(versionElement, "tags"));
+            // Repository
+            await UpdateMetadataRepository(metadata);
+            
+            // Dependencies
+            IList<PackageDependencyGroup> dependencyGroups = packageVersion.DependencySets.ToList();
+            if (dependencyGroups.Any())
+            {
+                metadata.Dependencies ??= new List<Dependency>();
 
-                    // licenses
-                    metadata.Licenses ??= new List<License>();
-                    metadata.Licenses.Add(new License()
-                    {
-                        Name = OssUtilities.GetJSONPropertyStringIfExists(versionElement, "licenseExpression"),
-                        Url = OssUtilities.GetJSONPropertyStringIfExists(versionElement, "licenseUrl")
-                    });
-                    
-                    // publishing info
-                    metadata.UploadTime = OssUtilities.GetJSONPropertyStringIfExists(versionElement, "published");
+                foreach (PackageDependencyGroup dependencyGroup in dependencyGroups)
+                {
+                    dependencyGroup.Packages.ToList().ForEach((dependency) => metadata.Dependencies.Add(new Dependency() { Package = dependency.ToString(), Framework = dependencyGroup.TargetFramework?.ToString()}));
                 }
```

Comment:
Nit: I wonder whether this could be simplified a bit:

```c#
metadata.Dependencies ??= dependencyGroups.SelectMany(group => group.Packages, (dependencyGroup, package) => new { dependencyGroup, package})
                                   .Select(dependencyGroupAndPackage => new Dependency() { Package = dependencyGroupAndPackage.package.ToString(), Framework = dependencyGroupAndPackage.dependencyGroup.TargetFramework?.ToString()})
                                   .ToList();
```

Or are we expecting `metadata.Dependencies` to possibly already have entries we want to keep?